src/software/backend/simulation/physics/physics_robot.cpp

```diff
@@ -24,119 +63,406 @@ PhysicsRobot::~PhysicsRobot()
     }
 }
 
-void PhysicsRobot::createRobotPhysicsBody(std::shared_ptr<b2World> world,
-                                          const Robot& robot)
+void PhysicsRobot::setupRobotBodyFixtures(const Robot& robot, double total_chicker_depth,
+                                          double mass_kg)
 {
-    // All the BodyDef must be defined before the body is created.
-    // Changes made after aren't reflected
-    robot_body_def.type = b2_dynamicBody;
-    robot_body_def.position.Set(robot.position().x(), robot.position().y());
-    robot_body_def.linearVelocity.Set(robot.velocity().x(), robot.velocity().y());
-    robot_body_def.angle           = robot.orientation().toRadians();
-    robot_body_def.angularVelocity = robot.angularVelocity().toRadians();
+    b2FixtureDef robot_body_fixture_def;
+    // This is a somewhat arbitrary value. Collisions with robots are not perfectly
+    // elastic. However because this is an "ideal" simulation and we generally don't care
+    // about the exact behaviour of collisions, getting this value to perfectly match
+    // reality isn't too important.
+    robot_body_fixture_def.restitution = 0.5;
+    robot_body_fixture_def.friction    = 0.0;
```

Comment:
nit: can we put these number in the header file

Reply:
done